allennlp/common/checks.py

```diff
@@ -107,10 +107,22 @@ def check_for_gpu(device_id: Union[int, List[int]]):
     elif device_id is not None and device_id >= 0:
         num_devices_available = cuda.device_count()
         if num_devices_available == 0:
+            # Torch will give a more informative exception than ours, so we want to include
+            # that context as well if it's available.  For example, if you try to run torch 1.5
+            # on a machine with CUDA10.1 you'll get the following:
+            #
+            #     The NVIDIA driver on your system is too old (found version 10010).
+            #
+            torch_gpu_error = ""
+            try:
+                cuda.current_device()
```

Comment:
```suggestion
                cuda._check_driver()
```

Under the hood the exception raised by `cuda.current_device()` comes from a call to `cuda._check_driver()`. Maybe we should just call that function directly? It's more clear to me that a function named `cuda._check_driver()` would raise an exception, whereas with `cuda.current_device()` I would think it just returns `None` if CUDA isn't available.

I mean.. I don't like using private functions from another lib, but since we're so closely coupled with PyTorch anyway, I think we might as well.

Reply:
Are you OK with my approach otherwise?  Works for me.

Reply:
Yeup! I think that's good